From 88057b1a7a519db5712b2ca71ed2e088250ef300 Mon Sep 17 00:00:00 2001
From: Harsh J <harsh@apache.org>
Date: Tue, 15 Jan 2013 14:13:16 +0000
Subject: [PATCH 1256/1357] MR1: MAPREDUCE-4930. Backport MAPREDUCE-4678 and MAPREDUCE-4925 to branch-1. Contributed by Karthik Kambatla and Chris McConnell. (harsh)

git-svn-id: https://svn.apache.org/repos/asf/hadoop/common/branches/branch-1@1433428 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit 3164be4d5b0f574058844cde984f72202a0d5293)

Reason: Customer request
Ref: CDH-8172
Author: Karthik Kambatla and Chris McConnell
---
 .../examples/dancing/DistributedPentomino.java     |   21 +++++++++++++++++++-
 1 files changed, 20 insertions(+), 1 deletions(-)

diff --git a/hadoop-mapreduce1-project/src/examples/org/apache/hadoop/examples/dancing/DistributedPentomino.java b/hadoop-mapreduce1-project/src/examples/org/apache/hadoop/examples/dancing/DistributedPentomino.java
index cef7063..218ae33 100644
--- a/hadoop-mapreduce1-project/src/examples/org/apache/hadoop/examples/dancing/DistributedPentomino.java
+++ b/hadoop-mapreduce1-project/src/examples/org/apache/hadoop/examples/dancing/DistributedPentomino.java
@@ -162,17 +162,36 @@ public class DistributedPentomino extends Configured implements Tool {
     int height = 10;
     Class<? extends Pentomino> pentClass;
     if (args.length == 0) {
-      System.out.println("pentomino <output>");
+      System.out
+          .println("Usage: pentomino <output> [-depth #] [-height #] [-width #]");
       ToolRunner.printGenericCommandUsage(System.out);
       return -1;
     }
     
     conf = new JobConf(getConf());
+
+    // Pick up the parameters, should the user set these
     width = conf.getInt("pent.width", width);
     height = conf.getInt("pent.height", height);
     depth = conf.getInt("pent.depth", depth);
     pentClass = conf.getClass("pent.class", OneSidedPentomino.class, Pentomino.class);
     
+    for (int i = 0; i < args.length; i++) {
+      if (args[i].equalsIgnoreCase("-depth")) {
+        depth = Integer.parseInt(args[++i].trim());
+      } else if (args[i].equalsIgnoreCase("-height")) {
+        height = Integer.parseInt(args[++i].trim());
+      } else if (args[i].equalsIgnoreCase("-width")) {
+        width = Integer.parseInt(args[++i].trim());
+      }
+    }
+
+    // Set parameters for MR tasks to pick up either which way the user sets
+    // them or not
+    conf.setInt("pent.width", width);
+    conf.setInt("pent.height", height);
+    conf.setInt("pent.depth", depth);
+
     Path output = new Path(args[0]);
     Path input = new Path(output + "_input");
     FileSystem fileSys = FileSystem.get(conf);
-- 
1.7.0.4

